onnx/defs/math/defs.cc

```diff
@@ -1473,6 +1473,62 @@ ONNX_OPERATOR_SET_SCHEMA(
 
           matmulShapeInference(ctx, 0, 1);
         }));
+static const char* CumSum_ver11_doc = R"DOC(
+Performs cumulative sum of the input elements along the given axis.
+By default, it will do the sum inclusively meaning the first element is copied as is.
+Through an `exclusive` attribute, this behavior can change to exclude the first element.
+It can also perform summation in the opposite direction of the axis. For that, set `reverse` attribute to 1.
+
+Example:
+```
+input_x = [1, 2, 3]
+axis=0
+output = [1, 3, 6]
+exclusive=1
+output = [0, 1, 3]
+exclusive=0
+reverse=1
+output = [6, 3, 1]
+exclusive=1
+reverse=1
+output = [3, 1, 0]
+```
+ )DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+      CumSum,
+      11,
+    OpSchema()
+            .SetDoc(CumSum_ver11_doc)
+      .Attr(
+          "exclusive",
+          "If set to 1 will return exclusive sum in which the top element is not included.",
+          AttributeProto::INT,
+          static_cast<int64_t>(0))
+      .Attr(
+          "reverse",
+          "If set to 1 will perform the sums in reverse direction.",
+          AttributeProto::INT,
+          static_cast<int64_t>(0))
+      .Input(0, "x", "An input tensor that is to be processed.", "T")
+      .Input(1, "axis", "(Optional) A 0-D tensor. Must be in the range [-rank(x), rank(x))", "tensor(int32)")
```

Comment:
need to support int64 too to be compatible with other ops.

Reply:
@liqunfu Are you talking about axis? Are you saying that we should have axis_int32 and axis_int64?
Could you give me an example of an OP that does this properly? thanks

Reply:
Ok i think what you're referring to. I found Slice to be a good example:
https://github.com/onnx/onnx/blob/master/docs/Operators.md#Slice
I'll change this spec to match that

Reply:
I did modify the spec. Please take a look

Reply:
I think having the type changed from “tensor(int32)” to "tensor(int64)" is good enough. don't have to support both. make sense?